game-core/src/main/java/games/strategy/triplea/ai/pro/util/ProSortMoveOptionsUtils.java

```diff
@@ -125,195 +115,126 @@ private ProSortMoveOptionsUtils() {}
         new ArrayList<>(unitAttackOptions.entrySet());
     list.sort(
         (o1, o2) -> {
+          final Collection<Territory> territories1 =
+              filterOutWinningTerritories(o1.getValue(), player, attackMap, calc);
+          final Collection<Territory> territories2 =
+              filterOutWinningTerritories(o2.getValue(), player, attackMap, calc);
 
           // Sort by number of territories that still need units
-          int numOptions1 = 0;
-          for (final Territory t : o1.getValue()) {
-            final ProTerritory patd = attackMap.get(t);
-            if (patd.getBattleResult() == null) {
-              patd.estimateBattleResult(calc, player);
-            }
-            if (!patd.isCurrentlyWins()) {
-              numOptions1++;
-            }
-          }
-          int numOptions2 = 0;
-          for (final Territory t : o2.getValue()) {
-            final ProTerritory patd = attackMap.get(t);
-            if (patd.getBattleResult() == null) {
-              patd.estimateBattleResult(calc, player);
-            }
-            if (!patd.isCurrentlyWins()) {
-              numOptions2++;
-            }
+          if (territories1.size() != territories2.size()) {
+            return territories1.size() - territories2.size();
           }
-          if (numOptions1 != numOptions2) {
-            return (numOptions1 - numOptions2);
-          }
-          if (numOptions1 == 0) {
+          if (territories1.size() == 0) {
```

Comment:
This line is failing the custom code check:

```
Avoid using size() to check for empty or not empty

Ex: size() == 0 -> isEmpty(), size() > 0 -> !isEmpty()

./game-core/src/main/java/games/strategy/triplea/ai/pro/util/ProSortMoveOptionsUtils.java:127:          if (territories1.size() == 0) {

[FIX]
```

Reply:
Done